### PR TITLE
[Network] BREAKING CHANGE: `az network dns zone`: Deprecate `--zone-type`, `registration-vnets` and `resolution-vnets`

### PR DESCRIPTION
**Pull request by rsh-msft**
**Related command**
az network dns zone update
az network dns zone create
az network dns zone list

**Description**
*  `az network dns zone (create|update)`: Arguments `--zone-type`, `--registration-vnets` and `--resolution-vnets`, which are deprecated are no longer supported.  
*  `az network dns zone update` uses the autogenerated command model, specific test coverage added.
*  `az network dns zone list` updated with autogenerated code, supports pagination arguments. 

* Corresponding aaz change: https://github.com/Azure/aaz/pull/444

**Testing Guide**
* Use this command to create a zone $zone in resource group $rg:
  `az network dns zone create -g $rg  -n $zone`
* Use this command to update an existing zone $zone in resource group $rg:
  `az network dns zone update-g $rg  -n $zone`
* Use this command to list upto $n zones in resource group $rg:
   `az network dns zone list -g $rg -o table --max-items $n`
* Use this command to list the next $n zones with the token from a previous list command.  
    `az network dns zone list -g $rg --next-token <token> --max-items $n`

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Network

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 1]
```diff
@@ -2488,26 +2488,14 @@ def create_dns_zone(cmd, resource_group_name, zone_name, parent_zone_name=None,
         add_dns_delegation(cmd, created_zone, parent_zone_name, resource_group_name, zone_name)
     return created_zone
 
+class DNSZoneUpdate(_DNSZoneUpdate):
```
Comment (djr-jsr) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:2491]: Either use register or use custom command, don't use both

File: src/azure-cli/azure/cli/command_modules/network/commands.py [hunk 2]
```diff
@@ -211,11 +205,12 @@ def load_command_table(self, _):
     from .operations.dns import DNSListReferences
     self.command_table["network dns list-references"] = DNSListReferences(loader=self)
 
-    with self.command_group('network dns zone', network_dns_zone_sdk) as g:
+    with self.command_group('network dns zone') as g:
+        from .custom import DNSZoneUpdate
```
Comment (djr-jsr) [inline src/azure-cli/azure/cli/command_modules/network/commands.py:209]: This is not needed if you register the command. It will show up on its own.

Comment (rsh-msft) [reply src/azure-cli/azure/cli/command_modules/network/commands.py]: Fixed, Removed

File: src/azure-cli/azure/cli/command_modules/network/custom.py (hunk 1 above)
Comment (rsh-msft) [reply src/azure-cli/azure/cli/command_modules/network/custom.py]: Fixed, Removed

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 3]
```diff
@@ -2459,56 +2459,20 @@ def add_dns_delegation(cmd, child_zone, parent_zone, child_rg, child_zone_name):
...
         "if_none_match": '*' if if_none_match else None,
         "location": 'global',
-        "tags": tags,
-        "zone_type": zone_type,
-        "resolution_virtual_networks": resolution_vnets_dict if resolution_vnets else None,
-        "registration_virtual_networks": registration_vnets_dict if registration_vnets else None
+        "tags": tags
     })
 
     if cmd.supported_api_version(min_api='2016-04-01', resource_type=ResourceType.MGMT_NETWORK_DNS) and parent_zone_name is not None:
```
Comment (djr-jsr) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:2471]: Can we remove this check? Are there any other places MGMT sdk is used?

**Review by djr-jsr — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/network/custom.py (hunk 3 above)
Comment (rsh-msft) [reply src/azure-cli/azure/cli/command_modules/network/custom.py]: Fixed, verified

**Comment by yonzhan** [comment]
Please fix CI issues

**Comment by necusjz** [comment]
do we need to update src/azure-cli/azure/cli/command_modules/network/aaz/latest/network/dns/zone/_create.py as well?

**Review by necusjz — APPROVED** [review]

**Pull request merged**